app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/StockStatusBaseSelectProcessor.php

```diff
@@ -28,24 +32,35 @@ class StockStatusBaseSelectProcessor implements BaseSelectProcessorInterface
      */
     private $stockStatusResource;
 
+    /**
+     * @var StockStatusConfigurableInterface
+     */
+    private $stockConfigurable;
+
     /**
      * @param StockConfigurationInterface $stockConfig
      * @param StockStatusResource $stockStatusResource
+     * @param StockStatusConfigurableInterface $stockConfigurable
      */
     public function __construct(
         StockConfigurationInterface $stockConfig,
-        StockStatusResource $stockStatusResource
+        StockStatusResource $stockStatusResource,
+        StockStatusConfigurableInterface $stockConfigurable = null
     ) {
         $this->stockConfig = $stockConfig;
         $this->stockStatusResource = $stockStatusResource;
+        $this->stockConfigurable = $stockConfigurable ?:
+            ObjectManager::getInstance()->get(StockStatusConfigurableInterface::class);
     }
 
     /**
-     * {@inheritdoc}
+     * @inheritdoc
      */
-    public function process(Select $select)
+    public function process(Select $select, int $productId): Select
```

Comment:
adding a new parameter to a public method breaks BC

Deprecate the method and add a new method with the new parameter(s) instead of adding them to a public method.
Reference the new method in a @see tag as a recommended replacement. Explain the reasons for replacing the old method with the new one (e.g., there is a bug in the old method).